### ISSUE TITLE
Llama Multipacking Correctness

### ISSUE THREAD

**Issue by Sanger2000** [axolotl-ai-cloud/axolotl#1375]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Multipacking should not attend to previous sub-sequences packed in the same overall sequence

### Current behaviour

I am not confident, but I suspect that for llama models with bs>1, the information about multiple examples being packed into a single sequence may be ignored, meaning that a given example may attend to all previous examples in the same sequence.

My lack of confidence is because I am not using axolotl, but rather re-used the axolotl monkeypatch for packing in my own training script when I encountered a cross-example contamination bug.

The offending lines would be here:
https://github.com/OpenAccess-AI-Collective/axolotl/blob/638c2dafb54f1c7c61a5f7ad40f8cf6965bec896/src/axolotl/monkeypatch/utils.py#L99-L155

and here:
https://github.com/OpenAccess-AI-Collective/axolotl/blob/638c2dafb54f1c7c61a5f7ad40f8cf6965bec896/src/axolotl/monkeypatch/llama_attn_hijack_flash.py#L481-L498

For bs>1 the cu_seqlens are incorrectly computed from the position ids that they are of dim>1. 

This results in us skipping that case, effectively throwing out/ignoring the `cu_seqlens` information when computing `cu_seqlens_q`.

### Steps to reproduce

Try passing in position ids from a bs > 1 to the first code block. This produces cu_seqlens with dim > 1, where the first dimension is the batch size

### Possible solution

Adding this line to the first fn fixed for me:
```
    if len(position_ids.shape) == 2:
        position_ids = position_ids.view(-1).unsqueeze(0)
```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11

### axolotl branch-commit

main/638c2dafb54f1c7c61a5f7ad40f8cf6965bec896

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
maybe @winglian can help clarify this

**Comment by mhenrichsen** [comment]
Closing as stale.
